### PR TITLE
Pbf pbem remember password

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Summary

- Add username/password to PBEM screen

After this update the PBEM screen 'play' button can be enabled by filling in all forms found only on the PBEM screen.

- Password handling on PBEM and PBF are updated to have 'remember password' buttons that will blank out the password on game exit if not selected. 

## Commits

commit a15d422e6343cb1323c9c79eb33ee9dade78a848

    Update PBF & PBEM to have username+password and remember password

commit 3db45c1fddbf8e8236303b39f0ff76001ba523db

    Improve help text formatting

commit 17b146c6f96417aeccec5bd00eb5a6039fc39b30

    Update handling of forum password
    
    - Do not store the password as a variable in view model
    - This makes it consistent with how we handle email password
      in the PBEM view model.
    - Of note, we set only a dummy text in the password field
    - When password field is set, we will set the ClientSetting
      password value and then toggle a flag to indicate password
      has been set, but we will not store it as a variable.

commit 7a98d1bc705d52d3eeb0b7b6561e87c3ee77ff18

    Make help text fetch be consistent remove the only if statement in ForumPosterEditor constructor

## Functional Changes

[] New map or map update
[] New Feature
[x] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[x] Problem fix:  https://github.com/triplea-game/triplea/issues/5921, https://github.com/triplea-game/triplea/issues/5856 
[] Other:   

## Testing
- test cover is pretty good, gives confidence on the backend
- play-tested the fields being set/unset to verify the label highlight and play button being enabled
- play-tested password remember being remembered and/or forgotton

### Before
Screenshot from 2020-04-05 20-10-48

### After
Screenshot from 2020-04-05 20-09-51

### Before
Screenshot from 2020-04-05 20-10-57

### After
Screenshot from 2020-04-05 20-06-31

### Before
Screenshot from 2020-04-05 20-11-10

### After
Screenshot from 2020-04-05 20-06-21

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+          + "Put your username and password for the AxisAndAllies.org forum into the "
+          + "username and password fields<br/>"
+          + "Click the Test Post button, to check that TripleA can reply to your forum "
+          + "post<br/>"
+          + "<i>Note:</i> Your AxisAndAllies.org username and password are not stored as part "
+          + "of the save game, but they are stored encrypted in the local file system "
+          + "if you select the option to remember your credentials. You may have to "
+          + "enter your username and password again if you open the save game on "
+          + "another computer."
+          + "</p></html>";
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java:53]: This is no longer accurate, the axis & allies forum migrated to NodeBB quite some time ago, so the description should be more or less equivalent to our forum

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          + "<b>Post:</b> The post of the smtp server (typically 25 for unencrypted, and 587 "
+          + "for TLS encrypted servers)<br/>"
+          + "Use TLS encryption: check this if yor server supports the TLS protocol for "
+          + "email encryption<br/>"
+          + "<i>Note:</i> Your SMTP login and password are not stored as part of the save game, "
+          + "but they are stored encrypted in the local file system if you select the "
+          + "option to remember your credentials.<br/>"
+          + "You may have to enter your login and password again if you open the save "
+          + "game on another computer."
+          + "</p></html>";
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java:85]: Also thoughts about moving this to a resource html file?
HTML in source code is alway kind of weird.

File: game-core/src/main/java/org/triplea/util/ExitStatus.java [hunk 3]
```diff
@@ -14,10 +14,10 @@
...
   FAILURE(1);
 
+  private static final Collection<Runnable> exitActions = new HashSet<>();
   private final int status;
-  private final List<Runnable> exitActions = new ArrayList<>();
 
-  public void addExitAction(final Runnable runnable) {
+  public static void addExitAction(final Runnable runnable) {
     exitActions.add(runnable);
   }
```
Comment (RoiEXLab) [inline game-core/src/main/java/org/triplea/util/ExitStatus.java:22]: Ok now there's no difference to https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Runtime.html#addShutdownHook(java.lang.Thread)
Thoghts about removing this code?

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java:53]: The help text could use improvement, it has not been given much love, just mainly restored so far. No PR so far has really sought to change or improve it.

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java (hunk 2 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/HelpTexts.java:85]: IMO it's overkill. A class with dedicated constants and lots of HTML IMO is fine, avoids complexity of making sure a resource file is in the right place. Eventually this is going to be behind a string identifier and presumably translated to multiple languages, I'd be concerned about code churn for anything short of that update.

File: game-core/src/main/java/org/triplea/util/ExitStatus.java (hunk 3 above)
Comment (DanVanAtta) [reply game-core/src/main/java/org/triplea/util/ExitStatus.java:22]: I didn't realize that API existed, good to learn about. It looks like we should indeed use that and remove our custom listener implementation.

**Pull request merged**